src/url.js

```diff
@@ -650,3 +650,17 @@ export function checkCorsUrl(url) {
 export function tryDecodeUriComponent(component, opt_fallback) {
   return tryDecodeUriComponent_(component, opt_fallback);
 }
+
+/**
+ * Adds the path to the given url.
+ *
+ * @param {!HTMLAnchorElement} a
+ * @param {string} url
+ * @param {string} path
+ * @return {string}
+ */
+export function appendPathToUrlWithA(a, url, path) {
+  a.href = url;
+  const pathname = a.pathname.replace(/\/?$/, '/') + path.replace(/^\//, '');
+  return a.protocol + '//' + a.host + pathname + a.search + a.hash;
```

Comment:
Can we replace `a.protocol + '//' + a.host` with `a.origin`?

Reply:
Yes, done!